src/platforms/mirserver/urlbuilderdispatcher.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2020 UBports Foundation
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3, as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+ * SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef URL_BUILDER_DISPATCHER_H
+#define URL_BUILDER_DISPATCHER_H
+
+#include <stdint.h>
+
+#include <QByteArray>
+#include <qpa/qplatformservices.h>
+#include <QLoggingCategory>
+
+#include "services.h"
+
+Q_DECLARE_LOGGING_CATEGORY(QTMIR_URLBUILDER)
+
+// The QtWayland url_open IPC call will split all URLs into 128-byte chunks then
+// call the event via Wayland IPC. This object un-splits the URL and sends it
+// to URL-Dispatcher once it receives the last chunk (denoted by 'remaining'
+// reaching zero).
+
+namespace qtmir
+{
+
+class URLBuilderDispatcher {
+public:
+    URLBuilderDispatcher(QPlatformServices*);
+
+    void url_input(uint32_t, std::string const&);
+
+private:
+    uint32_t mLastRemaining;
+    QByteArray mCurrentUrl;
+    QPlatformServices* mServices;
```

Comment:
One extra thing, since you create this new in qtwindowmanager please use a shared_ptr